src/main/java/com/teragrep/rlp_03/context/frame/access/Access.java

```diff
@@ -1,79 +1,44 @@
 package com.teragrep.rlp_03.context.frame.access;
 
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.Phaser;
 import java.util.function.Supplier;
 
 public final class Access implements Supplier<Lease> {
-
-    private long accessCount;
-    private boolean terminated;
-    private final Lock lock;
+    final Phaser phaser;
     public Access() {
-        this.accessCount = 0; // TODO consider using a semaphore
-        this.terminated = false;
-        this.lock = new ReentrantLock();
+        this.phaser = new Phaser(1);
```

Comment:
Add a note what the argument `1` is for as it is not immediately obvious

Reply:
add comment fix [c447182](https://github.com/teragrep/rlp_03/pull/84/commits/c44718257663e780cb7e9699b9c1159fcf9517e9)